sdk/src/integrationTest/java/TokenUpdateIntegrationTest.java

```diff
@@ -684,4 +688,1812 @@ void cannotUpdateNonFungibleTokenMetadataWhenMetadataKeyNotSet() throws Exceptio
+        var adminKey = PrivateKey.generateED25519();
+        var wipeKey = PrivateKey.generateED25519();
+        var kycKey = PrivateKey.generateED25519();
+        var freezeKey = PrivateKey.generateED25519();
+        var pauseKey = PrivateKey.generateED25519();
+        var supplyKey = PrivateKey.generateED25519();
+        var feeScheduleKey = PrivateKey.generateED25519();
+        var metadataKey = PrivateKey.generateED25519();
+
+        // Create a non-fungible token
+        var tokenId = Objects.requireNonNull(
+            new TokenCreateTransaction()
+                .setTokenName("Test NFT")
+                .setTokenSymbol("TNFT")
+                .setTokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+                .setTreasuryAccountId(testEnv.operatorId)
+                .setAdminKey(adminKey.getPublicKey())
+                .setWipeKey(wipeKey.getPublicKey())
+                .setKycKey(kycKey.getPublicKey())
+                .setFreezeKey(freezeKey.getPublicKey())
+                .setPauseKey(pauseKey.getPublicKey())
+                .setSupplyKey(supplyKey.getPublicKey())
+                .setFeeScheduleKey(feeScheduleKey.getPublicKey())
+                .setMetadataKey(metadataKey.getPublicKey())
+                .freezeWith(testEnv.client)
+                .sign(adminKey)
+                .execute(testEnv.client)
+                .getReceipt(testEnv.client)
+                .tokenId
+        );
+
+        var tokenInfoBeforeUpdate = new TokenInfoQuery()
+            .setTokenId(tokenId)
+            .execute(testEnv.client);
+
+        assertThat(tokenInfoBeforeUpdate.adminKey.toString()).isEqualTo(adminKey.getPublicKey().toString());
+        assertThat(tokenInfoBeforeUpdate.wipeKey.toString()).isEqualTo(wipeKey.getPublicKey().toString());
+        assertThat(tokenInfoBeforeUpdate.kycKey.toString()).isEqualTo(kycKey.getPublicKey().toString());
+        assertThat(tokenInfoBeforeUpdate.freezeKey.toString()).isEqualTo(freezeKey.getPublicKey().toString());
+        assertThat(tokenInfoBeforeUpdate.pauseKey.toString()).isEqualTo(pauseKey.getPublicKey().toString());
+        assertThat(tokenInfoBeforeUpdate.supplyKey.toString()).isEqualTo(supplyKey.getPublicKey().toString());
+        assertThat(tokenInfoBeforeUpdate.feeScheduleKey.toString()).isEqualTo(feeScheduleKey.getPublicKey().toString());
+        assertThat(tokenInfoBeforeUpdate.metadataKey.toString()).isEqualTo(metadataKey.getPublicKey().toString());
+
+        var emptyKeyList = new KeyList();
+
+        // Make the token immutable when updating all of its keys to an empty KeyList
+        // (trying to remove keys one by one to check all errors),
+        // signing with a key that is different from an Admin Key (implicitly with an operator key),
+        // and setting the key verification mode to NO_VALIDATION
+        assertThatExceptionOfType(ReceiptStatusException.class).isThrownBy(() -> {
```

Comment:
Extending the previous comment, this code block could be made as a function like this:
```java
 private static void testTokenUpdateTransactionFails(
            String tokenId,
            String keyType,
            Object key,
            TokenKeyValidation keyVerificationMode,
            TestEnv testEnv,
            Status expectedStatus
    ) {
        assertThatExceptionOfType(ReceiptStatusException.class).isThrownBy(() -> {
            TokenUpdateTransaction transaction = new TokenUpdateTransaction()
                .setTokenId(tokenId)
                .setKeyVerificationMode(keyVerificationMode);

            switch (keyType) {
                case "wipe":
                    transaction.setWipeKey(key);
                    break;
                case "kyc":
                    transaction.setKycKey(key);
                    break;
                case "freeze":
                    transaction.setFreezeKey(key);
                    break;
                case "pause":
                    transaction.setPauseKey(key);
                    break;
                case "supply":
                    transaction.setSupplyKey(key);
                    break;
                case "feeSchedule":
                    transaction.setFeeScheduleKey(key);
                    break;
                case "metadata":
                    transaction.setMetadataKey(key);
                    break;
                case "admin":
                    transaction.setAdminKey(key);
                    break;
                default:
                    throw new IllegalArgumentException("Invalid key type: " + keyType);
            }

            transaction.execute(testEnv.client).getReceipt(testEnv.client);
        }).withMessageContaining(expectedStatus.toString());
    }
```